tests/tests.ts

```diff
@@ -0,0 +1,172 @@
+import * as Promise from 'bluebird'
+import { expect } from 'chai'
+import * as fs from 'fs'
+import * as mocha from 'mocha'
+import * as tar from 'tar-stream'
+import * as path from 'path'
+
+import * as Resolve from '../src/index'
+import * as Utils from '../src/utils'
+
+// The following indices are mapped to the order of resolvers returned
+// by Resolve.getDefaultResolvers()
+// If the order that they are returned changes, then so should these indices
+// but that will be obvious because the tests will fail
+const dockerfileResolverIdx = 0
+const dockerfileTemplateResolverIdx = 1
+const archDockerfileResolverIdx = 2
+const defaultResolvers = Resolve.getDefaultResolvers()
+
+
+function getDockerfileFromTarStream(stream: NodeJS.ReadableStream): Promise<string> {
+  return new Promise<string>((resolve, reject) => {
+    const extract = tar.extract()
+    let foundDockerfile = false
+
+    extract.on('entry', (header: tar.TarHeader, stream: NodeJS.ReadableStream, next: () => void) => {
+      if (path.normalize(header.name) === 'Dockerfile') {
+        let contents = ''
+        stream.on('data', (data) => {
+          contents += data
+        })
+        stream.on('end', () => {
+          foundDockerfile = true
+          resolve(contents)
+        })
+      }
+      next()
+    })
+
+    extract.on('finish', () => {
+      if (!foundDockerfile) {
+        reject('Could not find a dockerfile in returned archive')
+      }
+    })
+    stream.pipe(extract)
+  })
+}
+
+describe('Resolvers', () => {
+  it('should return resolve a standard Dockerfile project', (done) => {
+    const stream = fs.createReadStream('./tests/test-files/Dockerfile/archive.tar')
+
+    const bundle = new Resolve.Bundle(stream, '', '')
+    Resolve.resolveBundle(bundle, defaultResolvers)
+    .then((resolved) => {
+      if (resolved.projectType !== defaultResolvers[dockerfileResolverIdx].name) {
+        done(new Error('Dockerfile resolver not used for Dockerfile project'))
+      } else {
+        done()
+      }
+    })
+    .catch(done)
```

Comment:
Oh of course, I was thinking that I because it needed further processing I should use the `done()` callback, but obviously just extending the promise chain makes much more sense.